### PR TITLE
system test: Add --log-driver test for podman run

### PR DESCRIPTION
**Pull request by ypu**
Add a test for testing different log driver in podman run

### CODE REVIEW

File: test/system/030-run.bats [hunk 1]
```diff
@@ -305,4 +305,22 @@ echo $rand        |   0 | $rand
...
+    run_podman run --log-driver journald --rm $IMAGE echo "podman log-driver test"
+    run grep  "podman log-driver test" <(journalctl)
+    is "$output" ".*podman log-driver test" "log should exist in journalctl output"
+
+    run_podman run --log-driver none --name test $IMAGE sleep 1
+    run_podman 125 logs test
+    is "$output" ".*this container is not logging output" "can not read log when set --log-driver to none"
+
+    run_podman run --log-driver json-file --rm $IMAGE sleep 1
+    is "$output" "" "json-file should also work"
```
Comment (rhatdan) [inline test/system/030-run.bats:323]: Could you report the first test to see that this actually works.

File: test/system/030-run.bats [hunk 2]
```diff
@@ -305,4 +305,22 @@ echo $rand        |   0 | $rand
     run_podman wait $cid
 }
 
+@test "podman run --log-driver" {
+    run_podman run --log-driver k8s-file --log-opt path=/var/log/podmantest.log --rm $IMAGE echo "podman log-driver test"
+    run grep "podman log-driver test" /var/log/podmantest.log
+    rm -f /var/log/podmantest.log
+    is "$output" ".*stdout.*podman log-driver test" "log should exist in log file"
+
+    run_podman run --log-driver journald --rm $IMAGE echo "podman log-driver test"
```
Comment (rhatdan) [inline test/system/030-run.bats:314]: Should generate random content so we are sure the new data is in the journal.

File: test/system/030-run.bats [hunk 3]
```diff
@@ -305,4 +305,22 @@ echo $rand        |   0 | $rand
...
+    run grep  "podman log-driver test" <(journalctl)
+    is "$output" ".*podman log-driver test" "log should exist in journalctl output"
```
Comment (rhatdan) [inline test/system/030-run.bats:316]: Could you also add the podman logs test here, to make sure we can read the data back from the journal via the comand.

File: test/system/030-run.bats [hunk 4]
```diff
@@ -305,4 +305,22 @@ echo $rand        |   0 | $rand
     run_podman wait $cid
 }
 
+@test "podman run --log-driver" {
+    run_podman run --log-driver k8s-file --log-opt path=/var/log/podmantest.log --rm $IMAGE echo "podman log-driver test"
+    run grep "podman log-driver test" /var/log/podmantest.log
```
Comment (rhatdan) [inline test/system/030-run.bats:310]: Could you use $PODMAN_TMPDIR for this tests to make sure we don't effect the system, or cause problems with multiple tests running at the same time.

**Comment by ypu** [comment]
Thanks @rhatdan  for your comments. Update the code but found some bugs in current version. The test will failed:
1. podman logs can not read from journald and w
2. --log-drover json-file will not store the logs to a file

Report an issue https://github.com/containers/podman/issues/7476 for record the problems.

**Comment by rhatdan** [comment]
LGTM
@ypu Thanks for the test.

**Comment by edsantiago** [comment]
This is (mostly) resolved in #7758, although that one does not test `--log-opt` nor journald.

**Comment by rhatdan** [comment]
@ypu Still working on this PR?

**Comment by ypu** [comment]
Hi @rhatdan Looks like @edsantiago already write another test for cover the --log-driver and the issue I report still not fixed yet. I'd like close this one now. And will add some check steps based on Ed's test after the issue is fixed.

Thanks @rhatdan and @edsantiago for all your review and comments.

**Pull request closed without merging**